sbndqm/dqmAnalysis/CRT/SBND/BernCRTdqmSBND_module.cc

```diff
@@ -339,182 +327,365 @@ if(!fragmentHandle.isValid() || fragmentHandle->size() == 0)
+          uint64_t t1ResetTDCDiff = fracTDCT1Reset > t1Reset[mac5] ? fracTDCT1Reset - t1Reset[mac5] : t1Reset[mac5] - fracTDCT1Reset;
+
+          if(fDebug) std::cout << "Sending metric T1ResetTDCDiff with value " << t1ResetTDCDiff << std::endl;
+          sbndaq::sendMetric("CRT_board", mac5Str, "T1ResetTDCDiff", t1ResetTDCDiff, 0, artdaq::MetricMode::LastPoint);
+        }
+
+      for(int ch = 0; ch < 32; ++ch)
+        {
+          std::string chStr = std::to_string(mac5*100 + ch);
+          if(fDebug) std::cout << "Sending metric ChReadoutRate with value " << chReadoutRate[mac5][ch] / fRateNormalisation << std::endl;
+          sbndaq::sendMetric("CRT_channel", chStr, "ChReadoutRate", chReadoutRate[mac5][ch] / fRateNormalisation, 0, artdaq::MetricMode::Average);
+        }
+    } // loop over mac5
+
+  /////////////////////////
+  // Event-Level Metrics //
+  /////////////////////////
   //Currently using "0" as my blank Mac5 address for the event-level metrics.
-  
+
+  if(boardsWithT0Reset > fBoardsRequiredForResetSpread)
+    {
+      uint64_t t0ResetSpread = t0ResetMax - t0ResetMin;
+      if(fDebug) std::cout << "Sending metric T0ResetSpread with value " << t0ResetSpread << std::endl;
+      sbndaq::sendMetric("CRT_event", "0", "T0ResetSpread", t0ResetSpread, 0, artdaq::MetricMode::Maximum);
+    }
+
+  if(boardsWithT1Reset > fBoardsRequiredForResetSpread)
+    {
+      uint64_t t1ResetSpread = t1ResetMax - t1ResetMin;
+      if(fDebug) std::cout << "Sending metric T1ResetSpread with value " << t1ResetSpread << std::endl;
+      sbndaq::sendMetric("CRT_event", "0", "T1ResetSpread", t1ResetSpread, 0, artdaq::MetricMode::Maximum);
+    }
+    
+  /*
   //Metrics which are on the Grafana:
   
- 	 //"CRT hits in beam window per plane per event"
-    for (int i=0;i<7;++i){
-    	if(debug) {std::cout<<"hitsperplane["<<i<<"]: "<<hitsperplane[i]<<std::endl;}
-        sbndaq::sendMetric("CRT_event", std::to_string(0),
-            std::string("CRT_hits_beam_plane_")+std::to_string(i),
-            hitsperplane[i],
-            0, artdaq::MetricMode::LastPoint);
-      }
-	  //"CRT T1 resets per event"
-	  if(debug) {std::cout<<"num_t1_resets: "<<num_t1_resets<<std::endl;}
+  //"CRT hits in beam window per plane per event"
+  for (int i=0;i<7;++i){
+    if(fDebug) {std::cout<<"hitsperplane["<<i<<"]: "<<hitsperplane[i]<<std::endl;}
     sbndaq::sendMetric("CRT_event", std::to_string(0),
-          "T1_resets_per_event",
-          num_t1_resets,
-          0, artdaq::MetricMode::LastPoint);
+		       std::string("CRT_hits_beam_plane_")+std::to_string(i),
+		       hitsperplane[i],
+		       0, artdaq::MetricMode::LastPoint);
+  }
+  //"CRT T1 resets per event"
+  if(fDebug) {std::cout<<"num_t1_resets: "<<num_t1_resets<<std::endl;}
+  sbndaq::sendMetric("CRT_event", std::to_string(0),
+		     "T1_resets_per_event",
+		     num_t1_resets,
+		     0, artdaq::MetricMode::LastPoint);
```

Comment:
Ah let's remove this if we're not using - git history will let us get it back if we want.

Reply:
Agreed - mainly left that as a sneaky way to highlight the event metrics 😅

Reply:
also interesting to note: "metrics that are on the Grafana"... If we do want to re-implement anything it's easier to just remake code